server/player/player.go

```diff
@@ -2641,10 +2651,11 @@ func (p *Player) Data() Data {
 			OffHand:      offHand,
 			MainHandSlot: p.heldSlot.Load(),
 		},
-		Effects:      p.Effects(),
-		FireTicks:    p.fireTicks.Load(),
-		FallDistance: p.fallDistance.Load(),
-		Dimension:    p.World().Dimension().EncodeDimension(),
+		EnderChestInventory: p.enderChest.Items(),
```

Comment:
I think this should use `p.enderChest.Slots()`. `p.enderChest.Items()` only returns non-zero item stacks and it doesn't retain the association of these stacks with their corresponding slot.

Reply:
Indeed it should